tensorflow/tsl/platform/cpu_info.cc

```diff
@@ -345,6 +352,89 @@ void InitCPUIDInfo() {
 
 #endif  // PLATFORM_IS_X86
 
+#ifdef PLATFORM_IS_ARM64
+
+class CPUIDInfo;
+void InitCPUIDInfo();
+
+CPUIDInfo* cpuid = nullptr;
+
+// Structure for basic CPUID info.
+class CPUIDInfo {
+ public:
+  CPUIDInfo() : implementer_(0), variant_(0), cpunum_(0) {}
+
+  static void Initialize() {
+    // Initialize cpuid struct.
+    if (cpuid != nullptr) {
+      return;
+    }
+
+    cpuid = new CPUIDInfo;
+
+    if (!(getauxval(AT_HWCAP) & HWCAP_CPUID)) {
+      return;
+    }
+
+    std::ifstream CPUspresent;
+    CPUspresent.open("/sys/devices/system/cpu/present", std::ios::in);
```

Comment:
Is this always going to work?  This seems like it would be very platform-dependent.

Reply:
Thanks @cantonios for the comment. Yes, as you have said it is platform dependent. I have made updates in 528c1dc to address this. For platforms that pass through `ifdefs` if file doesn't exists we will simply just revert to default behaviour as will not be picking up heuristics.